feat: add view recurring deposit product page

muskankhedia opened the pull request:
## Description

- Add View Recurring Deposit Product Page

## Related issues and discussion
#467 

## Screenshots, if any
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/36980003/71825248-77813800-30c1-11ea-8d85-ee12fe3915b9.gif)

**Edit 1:**
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/36980003/80418144-8e5bcc80-88f4-11ea-8e86-f1f1c74dbaba.gif)

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

muskankhedia commented:
@abhaychawla Please have a look at it and suggest any changes required.

muskankhedia commented:
@abhaychawla it's actually been too long. I think you are busy with some other works, can you please review the PR when possible?

Rishabh106 commented:
> @abhaychawla it's actually been too long. I think you are busy with some other works, can you please review the PR when possible?

Since GCI is on priority currently, he has decided to review the PRs of GCI students first and then the others after the period of GCI ends.

muskankhedia commented:
Yaa.. I have a idea on that. No problems 😄

spider0061 commented on src/app/products/recurring-deposit-products/recurring-deposit-product.resolver.ts:
```diff
@@ -0,0 +1,31 @@
+  constructor(private productsService: ProductsService) { }
+
+  /**
+   * Returns the loan product data.
```
Returns the recurring deposit product data.

spider0061 commented on src/app/products/recurring-deposit-products/recurring-deposit-product.resolver.ts:
```diff
@@ -0,0 +1,31 @@
+import { ProductsService } from '../products.service';
+
+/**
+ * Loan Product data resolver.
```
Recurring Product data resolver.

abhaychawla commented on src/app/products/recurring-deposit-products/view-recurring-deposit-product/view-recurring-deposit-product.component.html:
```diff
@@ -0,0 +1,415 @@
+          <span fxFlex="60%">{{ recurringDepositProduct.shortName }}</span>
+        </div>
+
+        <h3 class="mat-h3" fxFlexFill>Items</h3>
```
Please check the preview of http://localhost:4200/products/fixed-deposit-products/create as an example and accordingly update this into the correct sections. Check for any changes to the display like for Deposit Amount and Interest Rate Chart and update that as well. Take reference from View Loan/Saving/Share product as well so we can make the code consistent with the web app.

muskankhedia replied:
I have updated all the sections making the reference as provided above.

abhaychawla commented on src/app/products/recurring-deposit-products/view-recurring-deposit-product/view-recurring-deposit-product.component.html:
```diff
@@ -0,0 +1,415 @@
+
+        <div fxFlexFill>
+          <span fxFlex="40%">Maximum Deposit Amount:</span>
+          <span fxFlex="60%">{{ recurringDepositProduct.maxDepositAmount | number}}</span>
```
Some fields might not be mandatory during creation. They should not be displayed if they do not contain a value. Check the API of recurring deposit products and existing code implementation in the community app to know about all those fields and update the code accordingly.

muskankhedia replied:
I have checked for all the optional and compulsory fields in the API docs and updated accordingly. Please have a look.

abhaychawla commented on src/app/products/recurring-deposit-products/view-recurring-deposit-product/view-recurring-deposit-product.component.html:
```diff
@@ -0,0 +1,415 @@
+
+        <mat-divider [inset]="true"></mat-divider>
+
+        <div fxFlexFill>
+          <span fxFlex="60%" style="float:right;">
+            <a
+              href="#/interestratecharts/{{recurringDepositProduct.id}}/{{recurringDepositProduct.name}}/recurringdepositproduct">View
+              / Edit Interest Rate Charts</a>
+          </span>
+        </div>
```
This is not working properly. Instead of this please update the display as specified above.

muskankhedia replied:
Earlier the page wasn't implemented as it was common for the fixed deposit product and recurring deposit product. Now I have created a new component and implemented so that can also be used for Fixed Deposit Product. Please have a look and let me know if any changes required.

abhaychawla commented on src/app/products/recurring-deposit-products/view-recurring-deposit-product/view-recurring-deposit-product.component.scss:
```diff
@@ -0,0 +1,25 @@
+.margin-t {
+margin-top: 1em;
+}
+  
```
Only a single line should be present at the end of file.

abhaychawla commented on src/app/products/recurring-deposit-products/view-recurring-deposit-product/view-recurring-deposit-product.component.html:
```diff
@@ -0,0 +1,415 @@
+
+  </mat-card>
+
+</div>
```
Single line should be present at the end of file.

spider0061 commented:
@muskankhedia Squash your commits into one using git rebase -i HEAD~N. If you have any problem regarding rebasing you can read this article:
https://www.internalpointers.com/post/squash-commits-into-one-git

muskankhedia commented on src/app/products/recurring-deposit-products/view-recurring-deposit-product/view-recurring-deposit-product.component.ts:
```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
```
The dailog component has been removed and the screen has been updated with the Incentives table. Please have a look at the GIF attached in the description to have a look at the UI implemented. Let me know if any further changes required.

abhaychawla replied:
No, don't create a separate component for the interest rate chart. Follow the approach as in the Preview step of Fixed Deposit Component. (when New Fixed Deposit Product is created in the web-app)

muskankhedia replied:
@abhaychawla As much as I have understood, should I implement the edit option on clicking the `Edit Button` for the interest rate chart in the same page(view recurring deposit product page) instead of navigating to the new page. The table will be hidden and the slabs will be displayed as editable forms and also an option to add a new slab will be available.

abhaychawla replied:
For the time being, we will not have this separate edit option now for interest rate chart while viewing the product, we will only have it during the editing of the product. So no need for that. Just like the preview screen of Create Fixed Deposit Product make sure the interest rate chart slabs and incentives are displayed in table here without any option for editing them.

muskankhedia replied:
I got it. I will update it.

muskankhedia commented:
> Have a look at the review. Also, check the description of your commit message and update it properly. As per the commit message guidelines please include `Fixes` in the commit message along with the reference to the issue id as the PR will completely fix the issue. Once you update the PR, make sure the conflicts have also been resolved.

I have worked on all the comments and implemented the changes as mentioned. I have also fixed the merge conflicts and squashed the commits with a proper description. Please have a look.

abhaychawla commented on src/app/products/recurring-deposit-products/view-recurring-deposit-product/view-recurring-deposit-product.component.html:
```diff
@@ -0,0 +1,480 @@
+
+        <mat-divider [inset]="true"></mat-divider>
+
+        <div fxFlexFill *ngIf="recurringDepositProduct.minDepositAmount">
+          <span fxFlex="40%">Minimum Deposit Amount:</span>
+          <span fxFlex="60%">{{ recurringDepositProduct.minDepositAmount | number}}</span>
+        </div>
+
+        <div fxFlexFill>
+          <span fxFlex="40%">Default Deposit Amount:</span>
+          <span fxFlex="60%">{{ recurringDepositProduct.depositAmount | number}}</span>
+        </div>
+
+        <div fxFlexFill *ngIf="recurringDepositProduct.maxDepositAmount">
+          <span fxFlex="40%">Maximum Deposit Amount:</span>
+          <span fxFlex="60%">{{ recurringDepositProduct.maxDepositAmount | number}}</span>
```
For some cases where we have min, max, and default values they have been combined. See the components of Loan, Savings, Shares products as examples and update to maintain consistency wherever necessary.

muskankhedia replied:
Yaa.. I have made changes accordingly.

muskankhedia commented:
@abhaychawla I have updated the PR with the description. Kindly have a look.

muskankhedia commented:
@abhaychawla I have updated all the requested changes. Please have a look.

abhaychawla commented:
Resolve the conflicts for review.

muskankhedia commented:
@abhaychawla Done. Please review.
